pkg/auth/cache/synchronized_access_cache.go

```diff
@@ -91,12 +93,16 @@ func (s *SynchronizedAccessCache) Synch(ctx context.Context) error {
 
 		// store in temp cache
 		for _, s := range ss {
-			c[s] = append(c[s], ns)
+			if _, exists := c[s]; !exists {
+				c[s] = sets.New(ns.Name)
+			} else {
+				c[s].Insert(ns.Name)
+			}
```

Comment:
if we rely on sets here, we can remove the `removeDuplicateSubjects` function and its invocation at L92